public/app/core/utils/richHistory.ts

```diff
@@ -329,6 +329,55 @@ export function notEmptyQuery(query: DataQuery) {
   return false;
 }
 
+export function filterQueriesBySearchFilter(queries: RichHistoryQuery[], searchFilter: string) {
+  const filteredQueries = queries.filter(query => {
+    const searchInComment = query.comment.includes(searchFilter);
+    const searchInQueries = query.queries.filter(query => {
+      const queryHasSearchedValue = Object.values(_.omit(query, 'datasource')).some(value =>
+        value.toString().includes(searchFilter)
+      );
+      return queryHasSearchedValue;
+    });
+    return searchInComment || searchInQueries.length > 0;
+  });
+
+  return filteredQueries;
+}
+
+export function filterQueriesByDataSource(queries: RichHistoryQuery[], listOfDatasourceFilters: string[]) {
+  const filteredQueries =
+    listOfDatasourceFilters && listOfDatasourceFilters?.length > 0
+      ? queries?.filter(q => listOfDatasourceFilters?.includes(q.datasourceName))
+      : queries;
+
+  return filteredQueries;
+}
+
+export function filterQueriesByTime(queries: RichHistoryQuery[], timeFilter: [number, number]) {
+  const filteredQueries = queries?.filter(
```

Comment:
No need for `?`, since it's a required argument. (The same goes for everything in `filterQueriesByDataSource` above.)

Reply:
Resolved in https://github.com/grafana/grafana/pull/25747/commits/3313c6b0e430ceeb85beb17670c8b1215518229a